R/derive_params_growth_height.R

```diff
@@ -0,0 +1,282 @@
+#'
+#'
+#' @family der_prm_bds_vs
+#'
+#' @keywords der_prm_bds_vs
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AAGECUR,
+#'     new_var_unit = AAGECURU,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months"
+#'   )
+#'
+#' heights <- vs_peds %>%
+#'   filter(VSTESTCD == "HEIGHT") %>%
+#'   select(USUBJID, VSSTRESN, VSSTRESU, VSDTC) %>%
+#'   rename(
+#'     HGTTMP = VSSTRESN,
+#'     HGTTMPU = VSSTRESU
+#'   )
+#'
+#' advs <- advs %>%
+#'   right_join(., heights, by = c("USUBJID", "VSDTC"))
+#'
+#' advs_under2 <- advs %>%
+#'   filter(AAGECUR <= 24)
+#'
+#' advs_over2 <- advs %>%
+#'   filter(AAGECUR > 24)
+#'
+#' who_under2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_lgth_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_lgth_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT_LENGTH = Length,
+#'     HEIGHT_LENGTHU = height_unit
+#'   )
+#'
+#' who_over2 <- bind_rows(
+#'   (admiralpeds::who_wt_for_ht_boys %>%
+#'     mutate(
+#'       SEX = "M",
+#'       height_unit = "cm"
+#'     )
+#'   ),
+#'   (admiralpeds::who_wt_for_ht_girls %>%
+#'     mutate(
+#'       SEX = "F",
+#'       height_unit = "cm"
+#'     )
+#'   )
+#' ) %>%
+#'   rename(
+#'     HEIGHT_LENGTH = Height,
+#'     HEIGHT_LENGTHU = height_unit
+#'   )
+#'
+#'
+#' derive_params_growth_height(
+#'   advs_under2,
+#'   sex = SEX,
+#'   height = HGTTMP,
+#'   height_unit = HGTTMPU,
+#'   meta_criteria = who_under2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTHSDS",
+#'     PARAM = "Weight-for-height z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTHPCTL",
+#'     PARAM = "Weight-for-height percentile"
+#'   )
+#' )
+#'
+#' derive_params_growth_height(
+#'   advs_over2,
+#'   sex = SEX,
+#'   height = HGTTMP,
+#'   height_unit = HGTTMPU,
+#'   meta_criteria = who_over2,
+#'   parameter = VSTESTCD == "WEIGHT",
+#'   analysis_var = VSSTRESN,
+#'   set_values_to_sds = exprs(
+#'     PARAMCD = "WTHSDS",
+#'     PARAM = "Weight-for-height z-score"
+#'   ),
+#'   set_values_to_pctl = exprs(
+#'     PARAMCD = "WTHPCTL",
+#'     PARAM = "Weight-for-height percentile"
+#'   )
+#' )
+derive_params_growth_height <- function(dataset,
+                                        sex,
+                                        height,
+                                        height_unit,
+                                        meta_criteria,
+                                        parameter,
```

Comment:
@zdz2101 Can you please add HEIGHT_LENGTH and HEIGHT_LENGTHU as an argument to the function call as the metadata is now split for HEIGHT and LENGTH and we are calling the function separately for each parameter.  cc @rossfarrugia

Reply:
Adding reply from the template comment here:
sorry @Lina2689 @Fanny-Gautier at the team call today i realised i had incorrectly said above "you dont even need a HEIGHT_LENGTH variable" - i meant MEASURE here as we don't need any variable to say "Height" or "Length" if we no longer join these into one metadata file. We still should make the values and units consistent so that they work with the function from @zdz2101 (i.e. HEIGHT_LENGTH and HEIGHT_LENGTHU).

So I think the above can be updated on template side, rather than new function arguments.

Reply:
Updated in ADVS peds template directly. Thanks